core/src/main/scala/org/http4s/util/package.scala

```diff
@@ -12,34 +12,61 @@ package object util {
   def decode[F[_]](charset: Charset): Pipe[F, Byte, String] = {
     val decoder = charset.nioCharset.newDecoder
     val maxCharsPerByte = math.ceil(decoder.maxCharsPerByte().toDouble).toInt
-    val avgBytesPerChar = math.ceil(1.0 / decoder.averageCharsPerByte().toDouble).toInt
+    val avgBytesPerChar =
```

Comment:
If we're making changes here anyway, I guess we could roll back this whitespace formatting and not buy ourselves a bigger diff.